internal/wasm/store.go

```diff
@@ -511,7 +511,15 @@ func (s *Store) resolveImports(module *Module) (
 			importedFunction := imported.Function
 
 			actualType := importedFunction.Type
-			if !expectedType.EqualsSignature(actualType.Params, actualType.Results) {
+
+			var mismatch bool
+			if importedFunction.Kind != FunctionKindWasm {
+				// Host function takes two uint64 instead of a dedicated vector type in the signature.
+				mismatch = !expectedType.EqualsSignatureV128Flattened(actualType.Params, actualType.Results)
```

Comment:
So this seems weird special casing for host functions that want to take vector values as params or results ...

thoughts? @codefromthecrypt @anuraaga

Reply:
so basically with this we cannot judge, for example the host function `func myfunc(uint64, uint64)` should be imported for the signature either `(v128)` or `(i64, i64)` which we cannot disambiguate since we infer the host function's Wasm signature https://github.com/tetratelabs/wazero/blob/main/internal/wasm/gofunc.go#L240-L255

Reply:
though this is harmless :D